docs/types/reference-types.rst

```diff
@@ -133,8 +133,10 @@ It is possible to mark state variable arrays ``public`` and have Solidity create
 The numeric index becomes a required parameter for the getter.
 
 Accessing an array past its end causes a failing assertion. Methods ``.push()`` and ``.push(value)`` can be used
-to append a new element at the end of the array, where ``.push()`` appends a zero-initialized element and returns
-a reference to it.
+to append a new element at the end of a dynamically-sized array, where ``.push()`` appends a zero-initialized element and returns
+a reference to it. Dynamically-sized arrays can only be resized in storage. In memory such arrays can be of arbitrary 
+size but the size cannot be changed once an array is allocated.
+
```

Comment:
I think this would be better off as a note.

Also, please do line wrapping at the end of the sentence. This produces cleaner diffs because you don't have to rewrap everything around when you modify a single sentence.
```rst
to append a new element at the end of a dynamically-sized array, where ``.push()`` appends a zero-initialized element and returns
a reference to it.

.. note::
    Dynamically-sized arrays can only be resized in storage.
    In memory, such arrays can be of size but the size cannot be changed once an array is allocated.
```